src/main/java/com/epam/pages/popup/AcademicYearsAndVacationPopup.java

```diff
@@ -1,6 +1,112 @@
+    private WebElement errorMessageForWrongSelectedDates;
+    @FindBy(xpath = "//select[@class='ui-datepicker-year']/option")
+    private List<WebElement> listOfYears;
+
+    public boolean checkUIOfCreatePopupYearsSection() {
+        logger.info("Check UI of create popup in years section");
+        return uiHelper.checkElementsAreDisplayed(
+                title,
+                startDate,
+                endDate,
+                xButton,
+                saveButton
+        );
+    }
+
+    public boolean areDatesSelected() {
+        logger.info("Check if start date and end date are selected");
+        return uiHelper.areElementsSelected(
+                startDate,
+                endDate
+        );
+    }
+
+    public boolean checkErrorMessageForWrongSelectedDates() {
+        logger.info("Check if dates are selected wrong - start date < end date.");
+        return errorMessageForWrongSelectedDates.getText()
+                .equals(ErrorMessagesProvider.getWrongSelectedDatesErrMessage());
+    }
+
+    public boolean checkErrorMessageForForFillingLessThan30Days() {
+        logger.info("Check if dates are selected wrong - end date - start date < 30.");
+        return errorMessageForWrongSelectedDates.getText()
+                .equals(ErrorMessagesProvider.getLessThan30DaysErrMessage());
+    }
+
+    public void selectDate(int day, String month, int year, String field) {
+        logger.info("Fill {} date - {}/{}/{}", field, day, month, year);
+        if (field.contains("start")) {
+            uiHelper.clickOnWebElement(startDate);
+        } else if (field.contains("end")) {
+            uiHelper.clickOnWebElement(endDate);
+        }
+        new Select(yearToSelect).selectByValue(String.valueOf(year));
+        new Select(monthToSelect).selectByVisibleText(month);
+        selectDay(day).click();
+    }
+
+    public boolean isDateBeforeTodayEnabled() {
+        logger.info("Check if dates before today are enabled for selection.");
+        int day = LocalDate.now().getDayOfMonth();
+        uiHelper.clickOnWebElement(startDate);
+        return disabledDates.stream().allMatch(dates -> (Integer.parseInt(dates.getText()) < day)
+                && dates.getAttribute("class").contains("disabled"));
+    }
+
+    private WebElement selectDay(int day) {
+        return driver.findElement(By.xpath(String.format("//a[@data-date='%s']", day)));
+    }
+
+    public boolean isYearGraterThan10Years(int year) {
+        logger.info("Check if filled year is grater than 10 years from the moment of selection");
+        int startDateValue = SharedTestData.getStartDate().getYear();
+        return (year - startDateValue) > 10;
+    }
+
+    public boolean checkIfYearIsPresentInTheSelectList(int year) {
+        uiHelper.clickOnWebElement(endDate);
+        for (WebElement years : listOfYears) {
+            if (years.getText().equals(String.valueOf(year))) {
+                return true;
+            }
+        }
```

Comment:
Write this method using Java Streams, like the previous PR

Reply:
done